From 1ed26880cb376385f8ab3fd81300f3c3c837f616 Mon Sep 17 00:00:00 2001
From: Inseob Kim <inseob@google.com>
Date: Thu, 11 Mar 2021 13:06:45 +0900
Subject: [PATCH] Allow adbd to use vsock_socket

To allow microdroid's adbd connection to be forwarded, adbd should be
able to use vsock.

Bug: 181747352
Test: try to connect adb to microdroid after turning on selinux
Change-Id: Ia6662d5a028a82c8bbafa6c21da821e9a1144bdc
---
 private/adbd.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/private/adbd.te b/private/adbd.te
index 2c625652a..f569ad2fb 100644
--- a/private/adbd.te
+++ b/private/adbd.te
@@ -44,6 +44,9 @@ dontaudit adbd self:global_capability_class_set sys_resource;
 # this occurs. (b/123569840)
 dontaudit adbd self:{ socket vsock_socket } create;
 
+# Allow adbd inside vm to forward vm's vsock.
+allow adbd self:vsock_socket { create_socket_perms_no_ioctl listen accept };
+
 # Create and use network sockets.
 net_domain(adbd)
 
-- 
2.25.1

